### PR TITLE
Update NPM package to ES6

### PR DESCRIPTION
**Pull request by jasond-s**
add the travis yml

add all the outstanding query objects

add tests for usage

documentation and linting for standard

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,37 +1,124 @@
...
+// Import the module just like any other node dependancy.
+import codat from 'codat';
 
 var apiKey = '<YOUR API KEY HERE>';
-// Use codat.uat for UAT environment
-// Use codat.production for Production environment
 
-var codatApi = codat.uat(apiKey);
+// Use codat.uat for UAT environment.
+var codatApiUat = codat.uat(apiKey);
```
Comment (dhoare12) [inline README.md]: Is this still how you do this?

File: README.md [hunk 2]
```diff
@@ -1,37 +1,124 @@
...
+
+// Use codat.production for Production environment.
+var codatApi = codat.apiClient(codat.constants.UAT)(apiKey);
```
Comment (dhoare12) [inline README.md]: should this be codat.constants.PRODUCTION?

File: README.md [hunk 3]
```diff
@@ -1,37 +1,124 @@
...
+
+### <a name="22"></a> 2.2 Interacting with companies
+
+When you want to intereact directly with your linked companies you can use the helper method exposed by the API client.
+
+This set of features allows you to:
+
+1. `addCompany` - Add companies.
+2. `removeCompany` - Remove companies.
+2. `updateCOmpany` - Update companies.
```
Comment (dhoare12) [inline README.md]: typo

File: README.md [hunk 4]
```diff
@@ -1,37 +1,124 @@
...
+
+You can use the company data api client to build your own queries without the query objects and specifiy the arguments yourself. 
+
+In fact this is what the query objects use under the hood! 
+
+```javascript
+var companyId = 'ff36ff03-17de-47be-883a-5ceecbbc65ed';
+
+// You can also roll your own queries to the data api.
+// Be aware some query parameters are no availble on all endpoints.
```
Comment (dhoare12) [inline README.md]: typo

File: README.md (hunk 1 above)
Comment (jasond-s) [inline README.md]: Yep, have tried to make sure there aren't any breaking changes from the original readme.

File: README.md (hunk 2 above)
Comment (jasond-s) [inline README.md]: Yes.

File: README.md (hunk 3 above)
Comment (jasond-s) [inline README.md]: Yes

File: README.md (hunk 4 above)
Comment (jasond-s) [inline README.md]: Yes

**Pull request merged**